src/compile/legend.ts

```diff
@@ -167,29 +165,45 @@ namespace properties {
   }
 
   export function labels(fieldDef: FieldDef, symbolsSpec, model: UnitModel, channel: Channel): any {
+    let datum = '{{ datum.data }}';
+    let textTemplate = '{{ datum.data }}';
+
+    if(fieldDef.timeUnit) {
+      datum = '{{ datum.data | time:\'' + timeFormat(model, channel) + '\'}}'
+    }
+
+    if (fieldDef.unit) {
+      if (fieldDef.unitPosition == 'prefix') {
+        textTemplate = fieldDef.unit + datum;
+      } else if (fieldDef.unitPosition == 'suffix') {
+        textTemplate = datum + fieldDef.unit;
+      }
+    }
+
     if (channel === COLOR) {
       if (fieldDef.type === ORDINAL) {
         return {
           text: {
-            scale: model.scaleName(COLOR_LEGEND),
-            field: 'data'
+            template: textTemplate,
+            scale: model.scaleName(COLOR_LEGEND)
+            // field: 'data'
           }
         };
       } else if (fieldDef.bin) {
         return {
           text: {
-            scale: model.scaleName(COLOR_LEGEND_LABEL),
-            field: 'data'
-          }
-        };
-      } else if (fieldDef.timeUnit) {
-        return {
-          text: {
-            template: '{{ datum.data | time:\'' + timeFormat(model, channel) + '\'}}'
+            template: textTemplate,
+            scale: model.scaleName(COLOR_LEGEND_LABEL)
```

Comment:
This looks wrong. How does this add the unit to bins? I'd expect that this code generates labels like `20-30kg` instead of `20kg-30kg`.